lib/internal/modules/cjs/loader.js

```diff
@@ -806,7 +807,7 @@ Module.prototype._compile = function(content, filename) {
     result = compiledWrapper.call(thisValue, exports, require, module,
                                   filename, dirname);
   }
-  if (depth === 0) stat.cache = null;
+  if (requireDepth === 0) statCache = null;
   return result;
```

Comment:
I liked the symmetry of the previous approach where in the same block it inited a map/count and then after the operation nulled-the-map/decremented-the-count.

Reply:
I could move it back. It still behaves almost identical, it will just start caching entries a tiny bit earlier. That's why I changed it.

Reply:
Ya, if possible, the symmetry is nice and makes the code easier to follow.

Reply:
Do you feel strongly about this? Doing so is going to cache a bit less and we need two `statCache === null` checks instead of one (that was the case before as well but currently we only require one).

So I personally would rather keep it as is.

Reply:
:shipit: